src/components/SignUpForm/SignUpForm.js

```diff
@@ -0,0 +1,110 @@
+'use client'
+import React, { useEffect, useRef, useState } from "react";
+import { useRouter } from 'next/navigation'
+import styles from "./SignUpForm.module.css";
+import {CgShoppingBag} from 'react-icons/cg'
+import { getAuth, createUserWithEmailAndPassword,updateProfile, signOut } from "firebase/auth";
+import { collection,addDoc,setDoc,doc } from "@firebase/firestore";
+import Link from "next/link";
+import { db, auth} from "@/firebase/config";
+
+
+const SignUpForm = () => {
+    const [email, setEmail] = useState(null);
+    const [password, setPassword] = useState(null);
+    const [confirmPassword, setConfirmPassword] = useState(null);
+    const [name,setName] =useState(null)
+    const [error, setError] = useState(" ");
+    const router = useRouter();
+
+   
+    const submitHandler = async () =>{
+        /*const adding = await setDoc(doc(db, "users",122), 
+            userDetails
+        );*/
+        
+        if(confirmPassword != password){
+            setError("Error: (Passwords do not match)");
+        }else if(password.length <6){
+            setError("Error: (Weak password)");
```

Comment:
nit: maybe we can do some other password weakness check later, rather than just doing less than 6 characters

Reply:
Yes, I can try to avoid repeating characters in a sequence.